docs/reference/commandline/tye-cleanup.md

```diff
@@ -0,0 +1,70 @@
+# tye cleanup
+
+## Name
+
+`tye cleanup` - Removes a deployed application from Kubernetes.
+
+## Synopsis 
+
+```text
+tye cleanup [-?|-h|--help] [-i|--interactive] [-v|--verbosity <Debug|Info|Quiet>] [--what-if] [<PATH>]
+```
+
+## Description
+
+The `tye cleanup` command will delete a deployed application from Kubernetes. `tye cleanup` by default will:
+
+- List all resources that are part of an application
+- Print the list of resources (what-if)
+- Offer a choice delete each resource (interactive)
+- Delete each resource (if applicable)
+
+`tye cleanup` operates in the current Kubernetes namespace. Use `kubectl config view --minify --output 'jsonpath={..namespace}'` to view the current namespace.
+
+Cleanup decides which resources to delete based on the `app.kubernetes.io/part-of=...` label. This label will be set to the application name for all resources created by Tye. `tye cleanup` does not rely on the list of services in `tye.yaml` or a solution file. 
+
+## Arguments
+
+`PATH`
+
+The path to either a file or directory to execute `tye cleanup` on. Can either be a yaml, sln, or project file, however it is recommend to have a tye.yaml file for `tye cleanup`.
```

Comment:
Consider allowing service name, so someone can delete everything associated with a service

``` 
tye delete frontend
```

Reply:
I don't want to do service name because I think that raises a lot of questions - and we don't have per-service deployment yet.

For instance, does it update the config of every other service to remove its env-vars? (Probably not).

Per-application cleanup is easier to reason about because things like secrets/bindings are shared. Basically we *could* support per-service cleanup, but then what happens? does the usage of it by other services just dangle?

Reply:
That seems well justified.